src/apps/EDGE3D.cpp

```diff
@@ -0,0 +1,132 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-23, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "EDGE3D.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#include "AppsData.hpp"
+#include "common/DataUtils.hpp"
+
+#include <cmath>
+
+
+namespace rajaperf
+{
+namespace apps
+{
+
+
+EDGE3D::EDGE3D(const RunParams& params)
+  : KernelBase(rajaperf::Apps_EDGE3D, params)
+{
+  setDefaultProblemSize(100*100*100);  // See rzmax in ADomain struct
+  setDefaultReps(10);
+  constexpr int num_quadrature_points = NQ_1D*NQ_1D*NQ_1D;
+
+  Index_type rzmax = std::cbrt(getTargetProblemSize())+1;
+  m_domain = new ADomain(rzmax, /* ndims = */ 3);
+
+  m_array_length = m_domain->nnalls;
+  size_t number_of_elements = m_domain->lpz+1 - m_domain->fpz;
+
+  setActualProblemSize( number_of_elements );
+
+  setItsPerRep( number_of_elements );
+  setKernelsPerRep(1);
+
+  constexpr size_t matrix_size = NB*NB;
+
+  // touched data size, not actual number of stores and loads
+  // see VOL3D.cpp
+  size_t reads_per_node = 3*sizeof(Real_type);
+  size_t writes_per_rep = 1*sizeof(Real_type);
+  setBytesPerRep( writes_per_rep * getItsPerRep() +
+                  reads_per_node * (getItsPerRep() + 1+m_domain->jp+m_domain->kp) );
```

Comment:
I like the idea of using named variables here, big improvement in readability. I'll add an issue to refactor this for the rest of the perf suite.
```suggestion
  size_t reads_per_node = 3*sizeof(Real_type);
  size_t writes_per_zone = 1*sizeof(Real_type);
  setBytesPerRep( writes_per_zone * getItsPerRep() +
                  reads_per_node * (getItsPerRep() + 1+m_domain->jp+m_domain->kp) );
```